test/unittests/CandidateTest.php

```diff
@@ -244,6 +271,271 @@ public function testSetDataWithValueSucceeds()
+
+        $this->assertEquals(
+            $this->_candidateInfo['EDC'],
+            $this->_candidate->getCandidateEDC()
+        );
+    }
+
+    /**
+     * Test getCandidateSex returns the correct sex of the candidate
+     *
+     * @covers Candidate::getCandidateSex
+     * @return void
+     */
+    public function testGetCandidateSex()
+    {
+        $this->_setUpTestDoublesForSelectCandidate();
+        $this->_candidate->select($this->_candidateInfo['CandID']);
+
+        $this->assertEquals(
+            $this->_candidateInfo['Sex'],
+            $this->_candidate->getCandidateSex()
+        );
+    }
+
+    /**
+     * Test getCandidateEthnicity returns the correct ethnicity of the candidate
+     *
+     * @note   This function is deprecated so it was causing a PHAN error. 
+     *         It is marked test incomplete for this reason
+     *         however it is fully implemented and functional
+     * @covers Candidate::getCandidateEthnicity
+     * @return void
+     */
+    public function testGetCandidateEthnicity()
+    {
+        $this->markTestIncomplete(
+            "This test is marked as incomplete because the function is deprecated"
+        );
+        /**
+        $this->_setUpTestDoublesForSelectCandidate();
+        $this->_candidate->select($this->_candidateInfo['CandID']);
+
+        $this->assertEquals(
+            $this->_candidateInfo['Ethnicity'],
+            $this->_candidate->getCandidateEthnicity()
+        );
+        */
+    }
+
+    /** 
+     * Test isActive returns the correct string for the candidate
+     *
+     * @covers Candidate::isActive
+     * @return void
+     */
+    public function testIsActive()
+    {
+        $this->_setUpTestDoublesForSelectCandidate();
+        $this->_candidate->select($this->_candidateInfo['CandID']);
+
+        $this->assertEquals(
+            $this->_candidateInfo['Active'],
+            $this->_candidate->isActive()
+        );
+    }
+
+    /** 
+     * Test registeredBy returns the correct string for the candidate
+     *
+     * @covers Candidate::registeredBy
+     * @return void
+     */
+    public function testRegisteredBy()
+    {
+        $this->_setUpTestDoublesForSelectCandidate();
+        $this->_candidate->select($this->_candidateInfo['CandID']);
+
+        $this->assertEquals(
+            $this->_candidateInfo['RegisteredBy'],
+            $this->_candidate->registeredBy()
+        );
+    }
+
+    /**
+     * Test lastRecordChangeBy returns the correct UserID for the candidate
+     *
+     * @covers Candidate::lastRecordChangeBy
+     * @return void
+     */
+    public function testLastRecordChangeBy()
+    {
+        $this->_setUpTestDoublesForSelectCandidate();
+        $this->_candidate->select($this->_candidateInfo['CandID']);
+
+        $this->assertEquals(
+            $this->_candidateInfo['UserID'],
+            $this->_candidate->lastRecordChangeBy()
+        );
+    }
```

Comment:
shouldn't this change something and assert that the UserID changed too?

Reply:
As of now, the setData() method doesn't seem to change the UserID when something else is changed. So if I were to set the 'Sex' or something to a new value, the UserID would not change. I'm not sure where or how the UserID is changed once a change is made by a user, I'm assuming somehow in the frontend?

Reply:
That's a good question.. @ridz1208 @xlecours do either of you know how the UserID gets set in the candidate table?

Reply:
do we even have a way to change any of the values in the candidate from the front end ??

and if so which chnages should update the userId and which should not ? cause if I remember correctly some candidate parameters are stored in the candidate table. 

My guess is the usedId is only updated on creation

Reply:
Afaik, there isn't a way to change it from the front-end

Reply:
I think the UserID is supposed to be updated whenever that user makes a change. But in the Candidate.class.inc file, I'm not seeing anything that performs that action. There's also no setUserID or anything.